sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -0,0 +1,427 @@
+     *
+     * {@codesnippet com.azure.storage.blob.batch.BlobBatch.setBlobAccessTier#String-AccessTier-LeaseAccessConditions}
+     *
+     * @param blobUrl URI of the blob.
+     * @param accessTier The tier to set on the blob.
+     * @param leaseAccessConditions Lease access conditions that must be met to allow this operation.
+     * @return a {@link Response} that will be used to associate this operation to the response when the batch is
+     * submitted.
+     * @throws UnsupportedOperationException If this batch has already added an operation of another type.
+     */
+    public Response<Void> setBlobAccessTier(String blobUrl, AccessTier accessTier,
+        LeaseAccessConditions leaseAccessConditions) {
+        return setBlobAccessTierHelper(getUrlPath(blobUrl), accessTier, leaseAccessConditions);
+    }
+
+    private Response<Void> setBlobAccessTierHelper(String urlPath, AccessTier accessTier,
+        LeaseAccessConditions leaseAccessConditions) {
+        setBatchType(BlobBatchType.SET_TIER);
+        return createBatchOperation(blobAsyncClient.setAccessTierWithResponse(accessTier, null, leaseAccessConditions),
+            urlPath, EXPECTED_SET_TIER_STATUS_CODES);
+    }
+
+    private <T> Response<T> createBatchOperation(Mono<Response<T>> response, String urlPath,
+        int... expectedStatusCodes) {
+        int id = contentId.getAndIncrement();
+        batchOperationQueue.add(response
+            .subscriberContext(Context.of(BATCH_REQUEST_CONTENT_ID, id, BATCH_REQUEST_URL_PATH, urlPath)));
+
+        BlobBatchOperationResponse<T> batchOperationResponse = new BlobBatchOperationResponse<>(expectedStatusCodes);
+        batchMapping.put(id, batchOperationResponse);
+        return batchOperationResponse;
+    }
+
+    private String getUrlPath(String url) {
+        return UrlBuilder.parse(url).getPath();
+    }
+
+    private void setBatchType(BlobBatchType batchType) {
+        if (this.batchType == null) {
+            this.batchType = batchType;
+        } else if (this.batchType != batchType) {
+            throw logger.logExceptionAsError(new UnsupportedOperationException(String.format(Locale.ROOT,
+                "'BlobBatch' only supports homogeneous operations and is a %s batch.", this.batchType)));
+        }
+    }
+
+    Flux<ByteBuffer> getBody() {
+        if (batchOperationQueue.isEmpty()) {
+            throw logger.logExceptionAsError(new UnsupportedOperationException("Empty batch requests aren't allowed."));
+        }
+
+        // 'flatMap' the requests to trigger them to run through the pipeline.
+        Disposable disposable = Flux.fromStream(batchOperationQueue.stream())
+            .flatMap(batchOperation -> batchOperation)
+            .subscribe();
+
+        /* Wait until the 'Flux' is disposed of (aka complete) instead of blocking as this will prevent Reactor from
+         * throwing an exception if this was ran in a Reactor thread.
+         */
+        while (!disposable.isDisposed()) {
```

Comment:
This feels weird, runs in a tight loop, and could consume unnecessary CPU cycles.

What about:
```java
        return Flux.fromStream(batchOperationQueue.stream())
            .flatMap(batchOperation -> batchOperation)
            .then(Mono.fromRunnable(() -> {
                batchRequest.add(ByteBuffer.wrap(String.format("--%s--%s", batchBoundary, BlobBatchHelper.HTTP_NEWLINE).getBytes(StandardCharsets.UTF_8)));
            }))
            .thenMany(Flux.fromIterable(batchRequest));
```

Reply:
I've tried using this and it doesn't work for some reason. Given this is implementation detail and there is more work for batching next preview I'll write up an issue.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/5756